src/pages/Product.jsx

```diff
@@ -0,0 +1,146 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+
+export default class Product extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      evaluationForm: {
+        email: '',
+        message: '',
+        rating: '',
+      },
+      evaluations: [],
+    };
+
+    this.handleFormChange = this.handleFormChange.bind(this);
+    this.handleSubmitRating = this.handleSubmitRating.bind(this);
+  }
+
+  handleFormChange({ target }) {
+    const { name, value } = target;
+
+    this.setState(({ evaluationForm }) => {
+      const newEvaluationForm = { ...evaluationForm };
+      newEvaluationForm[name] = value;
+
+      return { evaluationForm: newEvaluationForm };
+    });
+  }
+
+  handleSubmitRating(e) {
+    e.preventDefault();
+
+    this.setState(({ evaluations, evaluationForm }) => {
+      const userEvaluation = { ...evaluationForm, id: evaluationForm.id };
```

Comment:
```suggestion
      const userEvaluation = { ...evaluationForm, id: evaluationForm.email };
```

como a gente discutiu sobre no discord